oap-server/server-library/library-elasticsearch-client/src/test/java/org/apache/skywalking/library/elasticsearch/ITElasticSearchTest.java

```diff
@@ -209,6 +226,15 @@ public void testSearch() {
             assertEquals(1, response.getHits().getTotal());
             assertEquals("val1", response.getHits().iterator().next().getSource().get("key1"));
         });
+        //test indexOnly
+        await().atMost(Duration.ONE_MINUTE).untilAsserted(() -> {
+            SearchResponse response = client.search(
+                Search.builder().query(Query.bool().must(Query.term("key3", "val3"))).build()
+            );
+            assertEquals(1, response.getHits().getTotal());
+            assertEquals("val1", response.getHits().iterator().next().getSource().get("key1"));
+            assertNull(response.getHits().iterator().next().getSource().get("key3"));
+        });
```

Comment:
Prefer an error message than comments so that when it failed we can see the reason from the logs directly

```suggestion
        await().atMost(Duration.ONE_MINUTE).untilAsserted(() -> {
            SearchResponse response = client.search(
                Search.builder().query(Query.bool().must(Query.term("key3", "val3"))).build()
            );
            assertEquals(1, response.getHits().getTotal());
            assertEquals("val1", response.getHits().iterator().next().getSource().get("key1"));
            assertNull("indexOnly fields should not be stored",
                       response.getHits().iterator().next().getSource().get("key3"));
        });
```

Reply:
Good suggestion!